static/app/components/sidebar/broadcasts.tsx

```diff
@@ -39,39 +39,34 @@ class Broadcasts extends Component<Props, State> {
 
   componentDidMount() {
     this.fetchData();
-
     document.addEventListener('visibilitychange', this.handleVisibilityChange);
   }
 
   componentWillUnmount() {
-    if (this.timer) {
-      window.clearTimeout(this.timer);
-      this.timer = null;
-    }
+    window.clearTimeout(this.markSeenTimeout);
+    window.clearTimeout(this.pollingTimeout);
 
-    if (this.poller) {
-      this.stopPoll();
-    }
     document.removeEventListener('visibilitychange', this.handleVisibilityChange);
   }
 
-  poller: number | null = null;
-  timer: number | null = null;
+  pollingTimeout: number | undefined = undefined;
+  markSeenTimeout: number | undefined = undefined;
 
-  startPoll() {
-    this.poller = window.setTimeout(this.fetchData, POLLER_DELAY);
+  startPolling() {
+    if (this.pollingTimeout) {
+      this.stopPolling();
+    }
+    this.pollingTimeout = window.setTimeout(this.fetchData, POLLER_DELAY);
   }
 
-  stopPoll() {
-    if (this.poller) {
-      window.clearTimeout(this.poller);
-      this.poller = null;
-    }
+  stopPolling() {
+    window.clearTimeout(this.pollingTimeout);
+    this.pollingTimeout = undefined;
   }
 
   fetchData = async () => {
-    if (this.poller) {
-      this.stopPoll();
+    if (this.pollingTimeout) {
+      this.stopPolling();
     }
```

Comment:
Probably safe enough to do
```suggestion
    this.stopPolling();
```